src/Microsoft.ML.Transforms/Text/WordEmbeddingsTransform.cs

```diff
@@ -0,0 +1,446 @@
+        {
+            Host.CheckValue(ctx, nameof(ctx));
+            ctx.CheckAtModel();
+            ctx.SetVersionInfo(GetVersionInfo());
+
+            SaveBase(ctx);
+            ctx.Writer.WriteBoolByte(_customLookup);
+            if (_customLookup)
+                ctx.SaveString(_modelFileNameWithPath);
+            else
+                ctx.Writer.Write((uint)_modelKind);
+        }
+
+        protected override ColumnType GetColumnTypeCore(int iinfo)
+        {
+            Host.Assert(0 <= iinfo && iinfo < Infos.Length);
+            return _outputType;
+        }
+
+        protected override Delegate GetGetterCore(IChannel ch, IRow input, int iinfo, out Action disposer)
+        {
+            Host.AssertValue(ch);
+            ch.AssertValue(input);
+            ch.Assert(0 <= iinfo && iinfo < Infos.Length);
+            disposer = null;
+
+            var info = Infos[iinfo];
+            if (!info.TypeSrc.IsVector)
+            {
+                throw Host.ExceptParam(nameof(input),
+                    "Text input given, expects a text vector");
+            }
+            return GetGetterVec(ch, input, iinfo);
+        }
+
+        private ValueGetter<VBuffer<float>> GetGetterVec(IChannel ch, IRow input, int iinfo)
+        {
+            Host.AssertValue(ch);
+            ch.AssertValue(input);
+            ch.Assert(0 <= iinfo && iinfo < Infos.Length);
+
+            var info = Infos[iinfo];
+            ch.Assert(info.TypeSrc.IsVector);
+            ch.Assert(info.TypeSrc.ItemType.IsText);
+
+            var srcGetter = input.GetGetter<VBuffer<DvText>>(info.Source);
+            var src = default(VBuffer<DvText>);
+            float[] wordVector = new float[_currentVocab.Dimension];
+
+            return
+                (ref VBuffer<float> dst) =>
+                {
+                    int deno = 0;
+                    srcGetter(ref src);
+                    var values = dst.Values;
+                    Utils.EnsureSize(ref values, 3 * _dimension, keepOld: false);
+                    Array.Clear(values, 0, 3 * _dimension);
+                    for (int i = 0; i < src.Count; i++)
+                        deno = FillOutputVector(ref src.Values[i], values, wordVector, deno);
+
+                    if (deno != 0)
+                        for (int index = 0; index < _dimension; index++)
+                            values[index + _dimension] /= deno;
+
+                    dst = new VBuffer<float>(values.Length, values, dst.Indices);
+                };
+        }
+
+        private int FillOutputVector(ref DvText src, float[] val, float[] wordVector, int deno)
+        {
+            if (_currentVocab.GetWordVector(ref src, wordVector))
+            {
+                deno++;
+                int offset = 2 * _dimension;
+                for (int i = 0; i < _dimension; i++)
+                {
+                    float currentTerm = wordVector[i];
+                    val[i] = val[i] < currentTerm ? val[i] : currentTerm;
```

Comment:
For correctness, I think we need a ` || i == 0`.

The Array.Clear will set the array to zeros, and we are only comparing to that value. This will incorrectly return 0.0 for the min/max in many cases.

Currently for, { 1, 2, 3, 4 }, we would return 0.0 for min.

Reply:
Can you point out the code change for this? I don't see it.